video/out/vulkan/context.c

```diff
@@ -186,27 +186,106 @@ pl_vulkan mppl_create_vulkan(struct vulkan_opts *opts,
     const char *opt_extensions[] = {
         VK_EXT_DESCRIPTOR_BUFFER_EXTENSION_NAME,
         VK_EXT_SHADER_ATOMIC_FLOAT_EXTENSION_NAME,
-#ifdef VK_EXT_SHADER_OBJECT_EXTENSION_NAME
-        VK_EXT_SHADER_OBJECT_EXTENSION_NAME,
-#endif
         VK_KHR_VIDEO_DECODE_QUEUE_EXTENSION_NAME,
         VK_KHR_VIDEO_DECODE_H264_EXTENSION_NAME,
         VK_KHR_VIDEO_DECODE_H265_EXTENSION_NAME,
         VK_KHR_VIDEO_QUEUE_EXTENSION_NAME,
-        "VK_KHR_video_decode_av1", /* VK_KHR_VIDEO_DECODE_AV1_EXTENSION_NAME */
-        "VK_KHR_video_maintenance2", /* VK_KHR_VIDEO_MAINTENANCE_2_EXTENSION_NAME */
+        /*
+         * Extensions below this point are newer than our minimum required Vulkan
+         * headers and so we only activate them if the build time headers contain
+         * them.
+         */
+#ifdef VK_EXT_shader_object
+        VK_EXT_SHADER_OBJECT_EXTENSION_NAME, /* 1.3.246 */
+#endif
+#ifdef VK_KHR_cooperative_matrix
+        VK_KHR_COOPERATIVE_MATRIX_EXTENSION_NAME, /* 1.3.255 */
+#endif
+#ifdef VK_KHR_video_maintenance1
+        VK_KHR_VIDEO_MAINTENANCE_1_EXTENSION_NAME, /* 1.3.274 */
+#endif
+        /*
+         * Because the AV1 extension does not require a feature flag to be enabled,
+         * we can include it as a string literal and have it work.
+         */
+        "VK_KHR_video_decode_av1", /* VK_KHR_VIDEO_DECODE_AV1_EXTENSION_NAME 1.3.277 */
+#ifdef VK_KHR_shader_relaxed_extended_instruction
+        VK_KHR_SHADER_RELAXED_EXTENDED_INSTRUCTION_EXTENSION_NAME, /* 1.3.288 */
+#endif
+#ifdef VK_KHR_video_maintenance2
+        VK_KHR_VIDEO_MAINTENANCE_2_EXTENSION_NAME, /* 1.4.306 */
+#endif
+#ifdef VK_KHR_video_decode_vp9
+        VK_KHR_VIDEO_DECODE_VP9_EXTENSION_NAME, /* 1.4.317 */
```

Comment:
Is this #ifdef madness really preferable to just listing them as strings?

Reply:
It is less error prone, and you need the #ifdef flood for the features.